### PR TITLE
[gardening]Remove commented code from URLSession 

### PR DESCRIPTION
**Pull request by nethraravindran**

### CODE REVIEW

File: Foundation/NSURLSession/EasyHandle.swift [hunk 1]
```diff
@@ -191,22 +188,10 @@ extension _EasyHandle {
     }
     /// Wait for pipelining/multiplexing
     /// - SeeAlso: https://curl.haxx.se/libcurl/c/CURLOPT_PIPEWAIT.html
-    //func set(waitForPipeliningAndMultiplexing flag: Bool) {
-    //    try! CFURLSession_easy_setopt_long(rawHandle, CFURLSessionOptionPIPEWAIT, flag ? 1 : 0).asError()
-    //}
```
Comment (pushkarnk) [inline Foundation/NSURLSession/EasyHandle.swift]: May make sense to document what the `CURLOPT_PIPEWAIT` option helps achieve here ..

File: Foundation/NSURLSession/EasyHandle.swift [hunk 2]
```diff
@@ -191,22 +188,10 @@ extension _EasyHandle {
...
     
-    /// set numerical stream weight
-    /// - Parameter weight: values are clamped to lie between 0 and 1
-    /// - SeeAlso: https://curl.haxx.se/libcurl/c/CURLOPT_STREAM_WEIGHT.html
-    /// - SeeAlso: http://httpwg.org/specs/rfc7540.html#StreamPriority
-    //func set(streamWeight weight: Float) {
-    //    // Scale and clamp such that the range 0->1 ends up 1->256
-    //    let w = 1 + max(0, min(255, Int(round(weight * 255))))
-    //    try! CFURLSession_easy_setopt_long(rawHandle, CFURLSessionOptionPIPEWAIT, w).asError()
-    //}
```
Comment (pushkarnk) [inline Foundation/NSURLSession/EasyHandle.swift]: May make sense to document what the CURLOPT_PIPEWAIT option helps achieve here ..

File: Foundation/NSURLSession/NSURLSessionTask.swift [hunk 3]
```diff
@@ -560,10 +556,8 @@ fileprivate extension URLSessionTask {
         easyHandle.set(followLocation: false)
         easyHandle.set(customHeaders: curlHeaders(for: request))
 
-        //Options unavailable on Ubuntu 14.04 (libcurl 7.36)
-        //TODO: Introduce something like an #if
-        //easyHandle.set(waitForPipeliningAndMultiplexing: true)
-        //easyHandle.set(streamWeight: priority)
+	//TODO: The CURLOPT_PIPEDWAIT option is unavailable on Ubuntu 14.04 (libcurl 7.36)
+	//TODO: Introduce something like an #if, if we want to set them here
```
Comment (pushkarnk) [inline Foundation/NSURLSession/NSURLSessionTask.swift]: May make sense to document what the CURLOPT_PIPEWAIT option helps achieve here ..

File: Foundation/NSURLSession/EasyHandle.swift [hunk 4]
```diff
@@ -167,9 +167,6 @@ extension _EasyHandle {
         try! CFURLSession_easy_setopt_long(rawHandle, CFURLSessionOptionPROTOCOLS, protocols).asError()
         try! CFURLSession_easy_setopt_long(rawHandle, CFURLSessionOptionREDIR_PROTOCOLS, protocols).asError()
         //TODO: Added in libcurl 7.45.0
-        // "https".withCString {
-        //     try! CFURLSession_easy_setopt_ptr(rawHandle, CFURLSessionOptionDEFAULT_PROTOCOL, UnsafeMutablePointer($0)).asError()
-        //}
     }
```
Comment (pushkarnk) [inline Foundation/NSURLSession/EasyHandle.swift]: Can we document, in a comment, what this code accomplished?

File: Foundation/NSURLSession/EasyHandle.swift [hunk 5]
```diff
@@ -167,9 +167,8 @@ extension _EasyHandle {
         try! CFURLSession_easy_setopt_long(rawHandle, CFURLSessionOptionPROTOCOLS, protocols).asError()
         try! CFURLSession_easy_setopt_long(rawHandle, CFURLSessionOptionREDIR_PROTOCOLS, protocols).asError()
         //TODO: Added in libcurl 7.45.0
-        // "https".withCString {
-        //     try! CFURLSession_easy_setopt_ptr(rawHandle, CFURLSessionOptionDEFAULT_PROTOCOL, UnsafeMutablePointer($0)).asError()
-        //}
+        //TODO: Set default protocol for schemeless URLs
+        //CURLOPT_ DEFAULT_PROTOCOL available only in libcurl 7.45.0
```
Comment (pushkarnk) [inline Foundation/NSURLSession/EasyHandle.swift]: Nit: please remove the space after CURLOPT_

**Comment by nethraravindran** [comment]
Cleaned up the commented code and addressed the changes

**Comment by pushkarnk** [comment]
@swift-ci please test and merge

**Comment by pushkarnk** [comment]
@swift-ci please test and merge

**Pull request merged**